### PR TITLE
cleanup: use _Static_assert for event table

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Uses:
* gcc4.6+ `_Static_assert` to assert the size of event_table
* designated initializers for whole event_table (C99)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/drivers/event_class/event_class.cpp [hunk 1]
```diff
@@ -8,14 +8,16 @@
...
-const char* cgroup_prefix_array[CGROUP_NUMBER] = {
+const char* cgroup_prefix_array[] = {
 	"cpuset=/",
 	"cpu=/",
 	"cpuacct=/",
 	"io=/",
 	"memory=/",
 };
 
+static_assert(sizeof(cgroup_prefix_array) / sizeof(*cgroup_prefix_array) == CGROUP_NUMBER, "Wrong number of cgroup_prefix_array.");
```
Comment (FedeDP) [inline test/drivers/event_class/event_class.cpp:19]: In cpp we have `static_assert` keyword.

**Review by Andreagit97 — APPROVED** [review]
/approve 
amazing :heart_eyes:

**Comment by Andreagit97** [comment]
uh we have conflicts :_(

**Comment by FedeDP** [comment]
Will fix them next week :)

**Comment by FedeDP** [comment]
Rebased :) @gnosek @Andreagit97

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by gnosek — APPROVED** [review]
<3

**Pull request merged**